xds/internal/balancer/balancergroup/balancergroup.go

```diff
@@ -132,13 +132,13 @@ func (sbc *subBalancerWithConfig) updateAddrs(addrs []resolver.Address) {
 		// This will be a common case with priority support, because a
 		// sub-balancer (and the whole balancer group) could be closed because
 		// it's the lower priority, but it can still get address updates.
-		return
+		return nil
 	}
 	if ub, ok := b.(balancer.V2Balancer); ok {
-		ub.UpdateClientConnState(balancer.ClientConnState{ResolverState: resolver.State{Addresses: addrs}})
-	} else {
-		b.HandleResolvedAddrs(addrs, nil)
+		return ub.UpdateClientConnState(s)
 	}
+	b.HandleResolvedAddrs(s.ResolverState.Addresses, nil)
+	return nil
```

Comment:
This pattern is more readable than the `if-else` pattern found in `startBalancer` and `updateSubConnState`. Could you change those.

Reply:
Done